js/apps/admin-ui/src/realm-settings/user-profile/AttributesGroupForm.tsx

```diff
@@ -110,82 +110,63 @@ export default function AttributesGroupForm() {
         titleKey={matchingGroup ? "editGroupText" : "createGroupText"}
         divider
       />
-      <PageSection variant="light" onSubmit={form.handleSubmit(onSubmit)}>
-        <FormAccess isHorizontal role="manage-realm">
-          <FormGroup
-            label={t("nameField")}
-            fieldId="kc-name"
-            isRequired
-            helperTextInvalid={t("required")}
-            validated={form.formState.errors.name ? "error" : "default"}
-            labelIcon={
-              <HelpItem helpText={t("nameHintHelp")} fieldLabelId="nameField" />
-            }
+      <PageSection variant="light">
+        <FormProvider {...form}>
+          <FormAccess
+            isHorizontal
+            role="manage-realm"
+            onSubmit={handleSubmit(onSubmit)}
           >
-            <KeycloakTextInput
-              id="kc-name"
-              isReadOnly={!!matchingGroup}
-              {...form.register("name", { required: true })}
+            <TextControl
+              name="name"
+              label={t("nameField")}
+              labelIcon={t("nameHintHelp")}
+              rules={{ required: t("required") }}
+              readOnlyVariant={matchingGroup ? "default" : undefined}
             />
-            {!!matchingGroup && (
-              <input type="hidden" {...form.register("name")} />
```

Comment:
What about this hidden input? I think it's there to prevent issues with the form not getting the right value, we might want to restore it perhaps (can be aside the `TextControl`). Alternatively, if this doesn't do anything let's remove it.

Reply:
Nah, less code is always better. Let's get rid of this.